### DISCUSSION TITLE
Building Applications

### DISCUSSION THREAD

**Discussion by adriankreiner** [microsoft/PlanetaryComputer#258] [Q&A] [answered]
Hi, 
What is the best way to build applications using a planetary computer (such as the Google Earth Engine)? 
Thanks a lot for any suggestions! 

Best, Adrian

**Comment by gilbertocamara** [comment]
If you are interested in land use/land cover, or agriculture mapping, consider using `sits`, an R package that provides an end-to-end solution for classification of Earth observation data cubes using MPC. The software is available on MPC, and the documentation is available on `https://e-sensing.github.io/sitsbook/`

**Comment by TomAugspurger** [comment, marked as answer]
I wouldn't say we have an officially recommended way. Because the Planetary Computer is built on open standards like STAC, you're free to use whatever software stack you.

That said, I would highlight our [`data` API](https://planetarycomputer.microsoft.com/docs/quickstarts/using-the-data-api/). Many applications will want to display one or more STAC items on a map. You can use the data API to do that, without having to deploy your own compute to read from Blob Storage (or do that client side in Javascript).

If you're building "data" applications / dashboards, I know that [Shiny](https://www.rstudio.com/products/shiny/) is popular in the R world. In the Python world there's Panel, Streamlit, Voila, Shiny, and probably others.

I'm less familiar with the Javascript world, but our [Data Catalog](https://planetarycomputer.microsoft.com/catalog) and [Explorer](https://planetarycomputer.microsoft.com/explore?c=30.0586%2C29.9930&z=2.00&v=2) could be considered applications building on the Planetary Computer's STAC API and data. https://github.com/microsoft/PlanetaryComputerDataCatalog/ contains the source for those.
